golang/cosmos/x/vstorage/keeper/querier_test.go

```diff
@@ -0,0 +1,107 @@
+package keeper
+
+import (
+	"bytes"
+	"testing"
+
+	agoric "github.com/Agoric/agoric-sdk/golang/cosmos/types"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func TestQuerier(t *testing.T) {
+	testKit := makeTestKit()
+	ctx, keeper := testKit.ctx, testKit.vstorageKeeper
+	legacyQuerierCdc := codec.NewAminoCodec(codec.NewLegacyAmino())
+	querier := NewQuerier(keeper, legacyQuerierCdc.LegacyAmino)
+
+	// Populate mock data
+	keeper.SetStorage(ctx, agoric.NewKVEntry("foo.bar", "42"))
+	keeper.SetStorage(ctx, agoric.NewKVEntry("foo.baz", "hello"))
+
+	type testCase struct {
+		label    string
+		path     []string
+		expected []byte
+		err      *sdkerrors.Error
+	}
+	testCases := []testCase{}
+
+	// Test invalid endpoint
+	testCases = append(testCases, []testCase{
+		{label: "invalid endpoint",
+			path: []string{"foo"},
+			err:  sdkerrors.ErrUnknownRequest,
+		},
+	}...)
+
+	// Test invalid arguments to valid data and children endpoints
+	for _, endpoint := range []string{"data", "children"} {
+		testCases = append(testCases, []testCase{
+			{label: endpoint + ": no entry path",
+				path: []string{endpoint},
+				err:  sdkerrors.ErrInvalidRequest,
+			},
+			{label: endpoint + ": too many segments",
+				path: []string{endpoint, "foo", "bar"},
+				err:  sdkerrors.ErrInvalidRequest,
+			},
+			{label: endpoint + ": invalid path",
+				path: []string{endpoint, ".", ""},
+				err:  sdkerrors.ErrInvalidRequest,
+			},
+		}...)
+	}
+
+	// Test data endpoint with valid vstorage path
+	testCases = append(testCases, []testCase{
+		{label: "data: non-existent path",
+			path: []string{"data", "bar"},
+			// DO NOT CHANGE
+			// The UI and CLI check for the specific cosmos-sdk error code & codespace
+			err: sdkerrors.ErrNotFound,
```

Comment:
Would it be going too far to also assert somewhere that  `sdkerrors.ErrNotFound` itself corresponds with code 6 in codespace "sdk"? That would capture our requirement of stability in cosmos-sdk.